src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -599,12 +599,16 @@ private static void DeleteTombstoneIfNeeded(DocumentsOperationContext context, C
             var tombstoneTable = context.Transaction.InnerTransaction.OpenTable(TombstonesSchema, collectionName.GetTableName(CollectionTableType.Tombstones));
             using (Slice.External(context.Allocator, lowerId, lowerSize, out Slice id))
             {
-                if (tombstoneTable.ReadByKey(id, out var reader) == false)
-                    return;
-
-                if (tombstoneTable.IsOwned(reader.Id))
+                foreach (var (tombstoneKey, tvh) in tombstoneTable.SeekByPrimaryKeyPrefix(id, Slices.Empty, 0))
```

Comment:
This will delete the existing values in the same collection, right?
But what about removing the postfix for the value when we delete the oldest value?

Reply:
Yes, this will delete the tombstone only if it in the given collection.

We don't care about removing the suffix, it is transparent to the other components since we always unwrap it (for replication, indexing, etl, etc.).